src/test/ui/parser/intersection-patterns.stderr

```diff
@@ -0,0 +1,33 @@
+error: pattern on wrong side of `@`
+  --> $DIR/intersection-patterns.rs:13:9
+   |
+LL |         Some(x) @ y => {}
+   |         -------^^^-
+   |         |         |
+   |         |         binding on the right, should be to the left
+   |         pattern on the left, should be to the right
+   |         help: switch the order: `y@Some(x)`
+
+error: left-hand side of `@` must be a binding pattern
+  --> $DIR/intersection-patterns.rs:23:9
+   |
+LL |         Some(x) @ Some(y) => {}
+   |         -------^^^-------
+   |         |         |
+   |         |         also a pattern
+   |         interpreted as a pattern, not a binding
+   |
+   = note: bindings are `x`, `mut x`, `ref x`, and `ref mut x`
+
+error: pattern on wrong side of `@`
+  --> $DIR/intersection-patterns.rs:32:9
+   |
+LL |         1 ..= 5 @ e => {}
+   |         -------^^^-
+   |         |         |
+   |         |         binding on the right, should be to the left
+   |         pattern on the left, should be to the right
+   |         help: switch the order: `e@1 ..=5`
```

Comment:
Can we make it space-separated?
```suggestion
   |         help: switch the order: `e @ 1 ..=5`
```

Reply:
Oh varkor has commented about this?

Reply:
We should, but see @varkor's comment in https://github.com/rust-lang/rust/pull/65410#discussion_r334546244. The bug is in pretty printing.